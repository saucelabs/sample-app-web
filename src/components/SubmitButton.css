.submit-button {
  background-color: #ffffff;
  border: 2px solid #e2231a;
  color: #e2231a;
  cursor: pointer;
  display: inline-block;
  outline: none;
  font-size: 18px;
  font-family: "Roboto", Arial, Helvetica, sans-serif;
  line-height: initial;
  margin-bottom: 15px;
  padding: 10px 30px;
  text-decoration: none;
  text-transform: uppercase;
  width: 100%;
}
