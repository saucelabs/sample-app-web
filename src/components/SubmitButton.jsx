import React from "react";
import PropTypes from "prop-types";
import "./SubmitButton.css";

const SubmitButton = ({ customClass, testId, value, ...props }) => {
  const extraClass = customClass ? ` ${customClass}` : "";
  return (
    <input
      type="submit"
      className={`submit-button${extraClass}`}
      value={value}
      {...(testId
        ? {
            "data-test": testId,
            id: testId,
            name: testId,
          }
        : {})}
      {...props}
    />
  );
};

SubmitButton.propTypes = {
  /**
   * A custom class
   */
  customClass: PropTypes.string,
  /**
   * The test id
   */
  testId: PropTypes.string,
  /**
   * The value of the input
   */
  value: PropTypes.string.isRequired,
};

SubmitButton.defaultProps = {
  customClass: undefined,
  testId: undefined,
};

export default SubmitButton;
