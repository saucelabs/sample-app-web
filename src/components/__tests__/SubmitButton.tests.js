import React from "react";
import { shallow } from "enzyme";
import SubmitButton from "../SubmitButton";

describe("SubmitButton", () => {
  it("should render with default props", () => {
    const props = {
      value: "Submit",
    };
    const component = shallow(<SubmitButton {...props} />);

    expect(component).toMatchSnapshot();
  });

  it("should render with a testId", () => {
    const props = {
      testId: "test-id",
      value: "Submit",
    };
    const component = shallow(<SubmitButton {...props} />);

    expect(component).toMatchSnapshot();
  });

  it("should render with a testId and a name prop", () => {
    const props = {
      testId: "test-id",
      value: "Submit",
      name: "name-id",
    };
    const component = shallow(<SubmitButton {...props} />);

    expect(component).toMatchSnapshot();
  });

  it("should render with custom props", () => {
    const props = {
      value: "Submit",
      foo: "bar",
      bar: "foo",
    };
    const component = shallow(<SubmitButton {...props} />);

    expect(component).toMatchSnapshot();
  });

  it("should render with a custom class", () => {
    const props = {
      customClass: "custom_class",
      value: "Submit",
    };
    const component = shallow(<SubmitButton {...props} />);

    expect(component).toMatchSnapshot();
  });
});
