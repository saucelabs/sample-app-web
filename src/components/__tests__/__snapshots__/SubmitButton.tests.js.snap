// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton should render with a custom class 1`] = `
<input
  className="submit-button custom_class"
  type="submit"
  value="Submit"
/>
`;

exports[`SubmitButton should render with a testId 1`] = `
<input
  className="submit-button"
  data-test="test-id"
  id="test-id"
  name="test-id"
  type="submit"
  value="Submit"
/>
`;

exports[`SubmitButton should render with a testId and a name prop 1`] = `
<input
  className="submit-button"
  data-test="test-id"
  id="test-id"
  name="name-id"
  type="submit"
  value="Submit"
/>
`;

exports[`SubmitButton should render with custom props 1`] = `
<input
  bar="foo"
  className="submit-button"
  foo="bar"
  type="submit"
  value="Submit"
/>
`;

exports[`SubmitButton should render with default props 1`] = `
<input
  className="submit-button"
  type="submit"
  value="Submit"
/>
`;
